pkg/util/instanceselector/instanceselector.go

```diff
@@ -137,9 +139,15 @@ func Setup(cloud, region, zone, defaultInstanceType string) error {
 	return nil
 }
 
-func getSelectorData(data, regionOrZone string) ([]InstanceData, error) {
+func getSelectorData(data, regionOrZone, filepath string) ([]InstanceData, error) {
 	d := make(map[string][]InstanceData)
-	err := json.Unmarshal([]byte(data), &d)
+	fileData, err := getSelectorDataFromFile(filepath, regionOrZone)
+	if err == nil {
+		// If loading from file is successful, return data from there,
+		// if not, fallback to data from instanceselector package
+		return fileData, nil
+	}
```

Comment:
We should log the error at a warning level to make it easier to debug if the user supplies an invalid file.

Reply:
:+1: added